### PR TITLE
Post caching

### LINKED ISSUE #221: Caching for posts

### PR DESCRIPTION
**Pull request by saraanej**
In this PR, I added caching for the posts. The list of posts in the feed activity are stored in the cache when there is an internet connection and when missing, the posts get fetched from the cache.

This closes #221

### CODE REVIEW

File: app/src/main/java/com/github/orkest/domain/persistence/AppEntities.kt [hunk 1]
```diff
@@ -26,5 +27,68 @@ class AppEntities {
...
+        return gson.toJson(date)
+    }
+
+    @TypeConverter
+    fun toOrkestDate(dateJson: String?): OrkestDate? {
+        val gson = Gson()
+        return gson.fromJson(dateJson, OrkestDate::class.java)
+    }
+}
+
```
Comment (evguenirousselot) [inline app/src/main/java/com/github/orkest/domain/persistence/AppEntities.kt:94]: For StringListConverter are you sure that the items in the list don't contain commas, because otherwise this won't work.

I think you can initialize GSON just once at the beginning of the class and then reuse it when needed.

File: app/src/main/java/com/github/orkest/ui/feed/FeedActivity.kt [hunk 2]
```diff
@@ -70,11 +63,52 @@ fun FeedActivity(viewModel: PostViewModel) {
...
+                postDescription = postEntity.postDescription,
+                song = postEntity.song,
+                nbLikes = postEntity.nbLikes,
+                likeList = postEntity.likeList,
+                nbComments = postEntity.nbComments,
+                media = postEntity.media,
+                isMediaVideo = postEntity.isMediaVideo
+            )
+        }
+    }
```
Comment (evguenirousselot) [inline app/src/main/java/com/github/orkest/ui/feed/FeedActivity.kt:111]: This could maybe lead to collisions : id = (0..100000).random()
Maybe autogenerate = true with @PrimaryKey

File: app/src/androidTest/java/com/github/orkest/PersistenceTesting.kt [hunk 3]
```diff
@@ -114,4 +130,16 @@ class MockSongDao: AppDao.Companion.SongDao {
...
+    private val postList = mutableListOf<AppEntities.Companion.PostEntity>()
+
+    override fun getAllPosts(): List<AppEntities.Companion.PostEntity> {
+        return postList
+    }
+
+    override fun insertPosts(posts: List<AppEntities.Companion.PostEntity>) {
+        postList.addAll(posts)
+    }
 }
```
Comment (evguenirousselot) [inline app/src/androidTest/java/com/github/orkest/PersistenceTesting.kt:145]: Very good tests, though do you think that considering more edge cases like null values or empty lists would have been interesting ?

**Review by evguenirousselot — APPROVED** [review]
Very good caching and great tests, nice work
